### PR TITLE
Feature/strategy pattern

### PR DESCRIPTION
**Pull request by luqp**

### CODE REVIEW

File: src/Bank/BankLogic/BankAccount.cs [hunk 1]
```diff
@@ -47,10 +56,15 @@ public bool HasEnoughCollateral(int amount)
             return amount > 0 && Balance >= amount * ratio;
         }
 
+        public int Fee()
```
Comment (uabarahona) [inline src/Bank/BankLogic/BankAccount.cs:59]: I am not seeing this method being used anywhere, why do we need it?

**Review by uabarahona — CHANGES_REQUESTED** [review]

File: src/Bank/BankLogic/BankAccount.cs (hunk 1 above)
Comment (luqp) [reply src/Bank/BankLogic/BankAccount.cs:59]: This is a new requirements of the bank, to charge a yearly maintenance fee of $5 for foreign

Comment (uabarahona) [reply src/Bank/BankLogic/BankAccount.cs:59]: Ok but the logic for change yearly fee is not implemented yet right?

Comment (luqp) [reply src/Bank/BankLogic/BankAccount.cs:59]: Yes, I only added the relation

**Review by uabarahona — APPROVED** [review]

**Pull request merged**